handler.py

```diff
@@ -59,19 +57,23 @@ def delete_message(group_id, message_id, token):
 
 
 def kick_user(group_id, user_id, token):
-    membership_id = get_membership_id(group_id, user_id, token)
+    membership_id, _ = get_membership_id_and_admin_status(group_id, user_id, token)
     remove_member(group_id, membership_id, token)
 
 
+
 def receive(event, context):
     message = json.loads(event['body'])
 
     bot_id = message['bot_id']
     for phrase in FLAGGED_PHRASES:
         if phrase in message['text'].lower():
-            kick_user(message['group_id'], message['user_id'], message['token'])
-            delete_message(message['group_id'], message['id'], message['token'])
-            send('Kicked ' + message['name'] + ' due to apparent spam post.', bot_id)
+            membership_id, is_admin = get_membership_id_and_admin_status(message['group_id'], message['user_id'], message['token'])
+            # Proceed with actions only if the user is not an admin
+            if not is_admin:
+                kick_user(message['group_id'], message['user_id'], message['token'])
+                delete_message(message['group_id'], message['id'], message['token'])
+                send('Kicked ' + message['name'] + ' due to apparent spam post.', bot_id)
```

Comment:
I'm ok with this, but do we want to send a message explaining that admins can't be kicked?

Reply:
If you want to implement that (I'd be supportive) feel free to open another PR

Reply:
I can't think of a reason we would ever want to kick an admin for saying ps5 and such. In my groups people said they would rather just not have any message at all for an admin .

Reply:
Maybe if people were playing around with it and were curious why some got banned and others didn't it would be useful. It's an easy addition either way